docs/release_notes.adoc

```diff
@@ -15,6 +15,7 @@ include::include.adoc[]
 * Add `globalTimeout` to  `@Timeout` extension, to apply a timeout to all features in a specification, configurable via the spock configuration file spockPull:1986[]
 * Add new <<extensions.adoc#default-value-provider,`IDefaultValueProviderExtension`>> extension point to add support for special classes in the Stub's default `EmptyOrDummyResponse` spockPull:1994[]
 * Add support for Groovy-4-style range expressions spockIssue:1956[]
+* Add `IStatelessAnnotationDrivenExtension` to allow a single extension instance to be reused across all specifications spockPull:2055[]
```

Comment:
Maybe add a sentence that some internal extensions where changed to be `IStatelessAnnotationDrivenExtension`, which could break if someone has subclassed them, although they are not part of the API.

Or maybe we should make the internal extension classes `final` that a user can't subclass them.

Reply:
> Or maybe we should make the internal extension classes final that a user can't subclass them.

I prefer to allow it, with the understanding that they might have to change their code when we update something.